src/permutation.rs

```diff
@@ -212,14 +212,14 @@ impl Permutation {
     // Uses a rayon multizip to allow more code flexibility while remaining
     // parallelizable. This can be adapted into a general product argument
     // for any number of wires.
-    pub(crate) fn compute_permutation_poly(
+    pub(crate) fn compute_permutation_vec(
         &self,
         domain: &EvaluationDomain,
         wires: [&[BlsScalar]; 4],
         beta: &BlsScalar,
         gamma: &BlsScalar,
         sigma_polys: [&Polynomial; 4],
-    ) -> Polynomial {
+    ) -> Vec<dusk_bls12_381::BlsScalar> {
```

Comment:
You can use `Vec<BlsScalar>` and don't have to mention the `dusk_bls12_381` module path since it is imported on the top of the file already.

More generally though, why do you not use `Polynomial` here? We have to convert the evals to to their polynomial form at some point, so why do not do it here?

Reply:
Agreed with @moCello

Reply:
the function `domain.ifft(w_vec)` requires a vector, we could compute it into the function but then in the prover, we need to treat each vector to be blinded in a different way. I think is more clear and clean to put the IFFT into the `blind_poly()`, and treat all inputs in the same way.